docs/design_proposals/user_survey_hooks.md

```diff
@@ -0,0 +1,135 @@
+# Title
+
+* Author(s): Tejal Desai
+* Design Shepherd: Brian de Alwis
+* Date: 07/11/2021
+* Status: Proposed
+
+## Background
+
+Currently, we get feedback on skaffold via skaffold HaTs survey. 
+However, in this survey we don't get feedback on new or existing features. 
+We also can't get enough information from our existing customers on how they are using certain features in case we want to make any changes.
+e.g.
+With render v2, we are changing how deploy works. This may affect our existing helm deployer users. 
+See [#6166](https://github.com/GoogleContainerTools/skaffold/issues/6166)
+## Design
+
+### User survey Definition
+In order to make sure only relevant surveys are shown to user, we have added a survey config.
+```
+type config struct {
+	id           string
+	promptText   string
+	expiresAt    time.Time
+	isRelevantFn func([]util.VersionedConfig) bool
+	link         string
+}
+
+```
+The survey config has two key fields
+1) expiresAt - This decided the lifetime of the survey. e.g. if we are collecting user feedback to get feedback on helm deployer re-design, then it makes sense to target the survey towards helm deployer user for a limited time until the re-design requirement phase is complete.
+2) isRelevantFn - This function determines if the user is a target audience for the survey. For the above example, it only makes sense to show the survey for helm deployer users, like this.
+```
+{
+  id: helmID,
+  expiresAt: time.Date(2021, time.August, 14, 00, 00, 00, 0, time.UTC),
+  isRelevantFn: func(cfgs []util.VersionedConfig) bool {
+	for _, cfg := range cfgs {
+		v1Cfg := cfg.(*latestV1.SkaffoldConfig)
+		if h := v1Cfg.Deploy.HelmDeploy; h != nil {
+			return true
+	        }
+        }
+	return false
+ },
+link: helmURL,
+},
+
+```
+something similar can be imaged for multi-module users
+```
+  isRelevantFn: func(cfgs []util.VersionedConfig) bool {
+	return len(cfgs) > 1
+ },
+```
+
+### How to prompt users to take user surveys
+
+#### Rules to show survey prompt
+When prompting users with surveys, we need to keep in mind 2 important rules
+1) Don't prompt users too often.
+   Currently, we only prompt users to fill in HaTs survey every two weeks until they fill.
+2) Don't prompt users if they have already taken the survey.
+   Currently, we only prompt users to fill in Hats Survey if they haven't taken it in last 3 months.
+
+For non HaTs surveys or user surveys, we will follow the same rules,
+1) Prompt users to fill in the survey once 2 weeks if its relevant to them until they fill it.
```

Comment:
Do you think we should revise this down to 1 week?  What if we have two surveys going (e.g., Helm and `test`).  Or will we prompt them for each possible survey?